### PR TITLE
CORS-3209: Set UserTags on CAPG resources

### PR DESCRIPTION
**Pull request by bfournie**
Set the UserTags from install-config.yaml on the machine and cluster resources in CAPG.

### CODE REVIEW

**Comment by bfournie** [comment]
To test, created the key as shown below
```
$ gcloud resource-manager tags keys list --parent=projects/145252386089
tagKeys/281480871530055  bobf-test-tag  Testing of tags with openshift-installer

$ gcloud resource-manager tags values list --parent=tagKeys/281480871530055
NAME                       SHORT_NAME  DESCRIPTION
tagValues/281475705299722  somevalue
```

To view the key attached to the worker node used the UI as below:
Screenshot from 2024-07-01 15-31-10

**Review by bharath-b-rh — COMMENTED** [review]
/lgtm

golint is failing because of the missing comment for the exported function `GetTagsFromInstallConfig`.

File: pkg/asset/manifests/gcp/cluster.go [hunk 1]
```diff
@@ -235,3 +235,20 @@ func getSubnet(ctx context.Context, project, region, subnetName string) (*comput
...
 	return subnet, nil
 }
+
+// GetTagsFromInstallConfig will return a slice of ResourceManagerTags from UserTags in install-config.
+func GetTagsFromInstallConfig(installConfig *installconfig.InstallConfig) []capg.ResourceManagerTag {
+	var tags []capg.ResourceManagerTag
+	if len(installConfig.Config.Platform.GCP.UserTags) > 0 {
+		tags = make([]capg.ResourceManagerTag, len(installConfig.Config.Platform.GCP.UserTags))
+		for i, tag := range installConfig.Config.Platform.GCP.UserTags {
+			tags[i] = capg.ResourceManagerTag{
```
Comment (barbacbd) [inline pkg/asset/manifests/gcp/cluster.go:245]: Are we doing any kind of validation here to ensure that the tags are meeting the capg (and probably gcp) requirements? I didn't see any in the installconfig/gcp/userTags.go file.

Comment (bfournie) [reply pkg/asset/manifests/gcp/cluster.go:245]: The userTags feature has the tags validation here - https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/gcp/usertags.go#L116
But you're right, its missing validations on the Tag fields, for example the key limits:
```
maxLength: 63
minLength: 1
pattern: ^[a-zA-Z0-9]([0-9A-Za-z_.-]{0,61}[a-zA-Z0-9])?$
```

I'm not sure we want to add those checks in this PRs, seems like it should be done in a separate PR, with separate bug, as its not related to CAPG specifically.

Comment (bfournie) [reply pkg/asset/manifests/gcp/cluster.go:253]: Note also that there is kubebuilder validations on the ResourceManagerTag so invalid values will be rejected by CAPG https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/api/v1beta1/tags.go#L34-L37

**Review by barbacbd — COMMENTED** [review]
/lgtm

File: pkg/asset/manifests/gcp/cluster.go [hunk 2]
```diff
@@ -242,3 +243,20 @@ func getSubnet(ctx context.Context, project, region, subnetName string) (*comput
 
 	return subnet, nil
 }
+
+// GetTagsFromInstallConfig will return a slice of ResourceManagerTags from UserTags in install-config.
+func GetTagsFromInstallConfig(installConfig *installconfig.InstallConfig) []capg.ResourceManagerTag {
+	var tags []capg.ResourceManagerTag
+	if len(installConfig.Config.Platform.GCP.UserTags) > 0 {
+		tags = make([]capg.ResourceManagerTag, len(installConfig.Config.Platform.GCP.UserTags))
```
Comment (r4f4) [inline pkg/asset/manifests/gcp/cluster.go:251]: Nit:
```suggestion
	var tags := make([]capg.ResourceManagerTag, len(installConfig.Config.Platform.GCP.UserTags))
```

Comment (bfournie) [reply pkg/asset/manifests/gcp/cluster.go]: Fixed.

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Review by barbacbd — COMMENTED** [review]
/approve

**Pull request merged**